src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj

```diff
@@ -14,18 +14,20 @@
   <ItemGroup>
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.DependencyInjection\src\Microsoft.Extensions.DependencyInjection.csproj" SkipUseReferenceAssembly="true" />
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.DependencyInjection.Abstractions\src\Microsoft.Extensions.DependencyInjection.Abstractions.csproj" SkipUseReferenceAssembly="true" />
-    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.DependencyInjection.Specification.Tests\src\Microsoft.Extensions.DependencyInjection.Specification.Tests.csproj"/>
+    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.DependencyInjection.Specification.Tests\src\Microsoft.Extensions.DependencyInjection.Specification.Tests.csproj" />
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
-    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="6.0.0" />
-    <PackageReference Include="DryIoc.Microsoft.DependencyInjection" Version="4.0.0" />
-    <PackageReference Include="Lamar.Microsoft.DependencyInjection" Version="2.1.0" />
-    <PackageReference Include="LightInject.Microsoft.DependencyInjection" Version="3.3.1" />
-    <PackageReference Include="StructureMap.Microsoft.DependencyInjection" Version="2.0.0" />
-    <PackageReference Include="Grace.DependencyInjection.Extensions" Version="7.0.1" />
-    <PackageReference Include="Stashbox.Extensions.Dependencyinjection" Version="3.1.1" />
-    <PackageReference Include="Unity.Microsoft.DependencyInjection" Version="2.1.3" />
+    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="7.2.0" />
+    <PackageReference Include="DryIoc.Microsoft.DependencyInjection" Version="5.1.0" />
+    <PackageReference Include="LightInject.Microsoft.DependencyInjection" Version="3.4.1" />
+    <PackageReference Include="Grace.DependencyInjection.Extensions" Version="7.1.0" />
+    <PackageReference Include="Stashbox.Extensions.Dependencyinjection" Version="3.2.1" />
   </ItemGroup>
 
+  <!-- These packages don't support .NETFramework -->
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
```

Comment:
>'$(TargetFrameworkIdentifier) [](http://example.com/codeflow?start=24&length=29)

similar condition written the following way up. should we update it too?

Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'"

Reply:
You can't because of the way MSBuild works. When you are evaluating properties, `$(TargetFrameworkIdentifier)` hasn't been set yet. So when you are evaluating properties you need to do it the more complicated way - as above in SignAssembly.

Reply:
Are you saying TargetFrameworkIdentifier is not set when evaluating the properties but it is set when evaluating the ItemGroup's? it seems strange to me :-)

Reply:
What can I say? MSBuild is strange. 🤷 

The way it works is first all properties are evaluated in the project - top to bottom. TargetFrameworkIdentifier is inferred from setting the TargetFramework property. This inference can only happen after the .csproj sets the TargetFramework property, which means it happens in a `.targets` file that is evaluated after the stuff inside the `.csproj`. Since TargetFrameworkIdentifier isn't set yet, you can't use it in a condition when evaluating properties.

After properties, all Items are evaluated - top to bottom. Since all properties are set by now, TargetFrameworkIdentifier is set and you can use it in a Condition.